### PR TITLE
Transferred shrinking functionality to edges

### PR DESCRIPTION
**Pull request by CodeFloh**
## Description ##

The already existing element shrinking functionality, which is mostly used for visualization purposes, has been moved from volumes to edges. It's now available for edges, faces, and volumes. An example (examples/show_quads.py) is provided.

The function used to create a face representation of a volume mesh. The new implementation always maintains the element type, i.e., for volumes it returns volumes. This might break some code since an additional face extraction may now be necessary in some cases.

### CODE REVIEW

File: examples/show_quads.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    quad = gus.Faces(vertices.vertices, connec)
+
+    # show
+    quad.show()
+    quad.shrink().show()
+
+    e = quad.shrink(ratio=0.7).toedges(False)
+    e.vis_dict["arrows"] = True
+    e.show()
+
```
Comment (j042) [inline examples/show_quads.py:20]: Regarding `show_quads.py`, I think this one shows more than just quads

**Review by j042 — CHANGES_REQUESTED** [review]
Good stuff!
Since this one is applicable from edges, you could perhaps emphasize that with:
```
import gustaf as gus
import numpy as np


if __name__ == "__main__":
    # create 2x3x4 test hexa element
    v_res = [2, 3, 4]
    vertices = gus.create.vertices.raster(
        bounds=[[0, 0, 0], [1, 1, 1]],
        resolutions=v_res
    )
    connec = gus.utils.connec.make_hexa_volumes(v_res)
    v = gus.Volumes(vertices.vertices, connec)

    try:
        import vedo # if nothing's raised, following should be usable

        # v shrink - f shrink - e shrink
        e = v.shrink().tofaces().shrink().toedges().shrink()
        e.vis_dict["arrows"] = True

        direct_toedges = v.toedges(unique=False).shrink()
        direct_toedges.vis_dict["arrows"] = True

        # not the most efficient way, but it is possible.
        gus.show.show_vedo(
            ["v, Volumes", v],
            ["v.shrink()", v.shrink()],
            ["v.shrink().tofaces().shrink()", v.shrink().tofaces().shrink()],
            [
                "v.shrink().tofaces().shrink().toedges().shrink()",
                v.shrink().tofaces().shrink().toedges().shrink(),
            ],
            ["as arrows - useful for orientation check!", e],
            ["v.toedges(unique=False).shrink()\nas arrows", direct_toedges]
        )

    except:
        v.show()
        v.shrink().tofaces().show()
        v.shrink().tofaces().shrink().show()
        v.shrink().tofaces().shrink().toedges().show()
        v.shrink().tofaces().shrink().toedges().shrink().show()
```

**Comment by j042** [comment]
> This might break some code since an additional face extraction may now be necessary in some cases.

Oh, good point. probably need to copy `vertexdata`/`vis_dict` at `tofaces()`, `toedges()`, `tovertices()`

File: examples/show_quads.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import numpy as np
+
+
+if __name__ == "__main__":
+    # create 4x5 element test mesh
+    v_res = [5, 6]
+    vertices = gus.create.vertices.raster(bounds=[[0, 0], [1, 1]], resolutions=v_res)
+    connec = gus.utils.connec.make_quad_faces(v_res)
+
+    quad = gus.Faces(vertices.vertices, connec)
```
[the remark this answers is not in the document]
Comment (CodeFloh) [reply examples/show_quads.py]: Will have a look at this in the other feature branch. ;)

File: examples/show_quads.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    # show
+    quad.show()
+    quad.shrink().show()
+
+    e = quad.shrink(ratio=0.7).toedges(False)
```
[the remark this answers is not in the document]
Comment (CodeFloh) [reply examples/show_quads.py]: Obsolete

File: examples/show_quads.py (hunk 1 above)
Comment (CodeFloh) [reply examples/show_quads.py]: I renamed the file.

**Comment by j042** [comment]
> I changed the example as per your suggestion but:
> 
> image
> 
> There is an issue with `tofaces()`. I.e., I don't know if the issue is with `tofaces()` but if it's just the way these elements are created. Any ideas about this? I will have a look when I create the `window` function.
 
I thought #14 fixed that problem. Did you rebase from the latest `main`?

**Review by j042 — APPROVED** [review]
good stuff

**Pull request merged**